### PR TITLE
Adding codespell as a spellchecker in the docs

### PR DESCRIPTION
**Pull request by SensibleSalmon**
ref #890 

I elected to leave aspell as an option.

Would it be worth adding a maketarget that clones codespell, does spellchecking and then wipes out the cloned directory?
- [x] Is it mergeable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Is the Copyright year up to date?

### CODE REVIEW

**Comment by ctb** [comment]
Is the PR title misspelling intentional? Or just ironic?

**Comment by bocajnotnef** [comment]
I elect that it's intentional.

**Comment by bocajnotnef** [comment]
@mr-c high bandwidth comments: use makefile from codespell to install, changing prefix a la https://github.com/mr-c/misc/wiki/Linux-Home-Directory-Program-Installation

e.g. `make --prefix=$HOME install`

**Comment by mr-c** [comment]
just `make prefix=${HOME}`

On Tue, Jul 28, 2015 at 2:04 PM Jake Fenton <notifications@github.com>
wrote:

> @mr-c <https://github.com/mr-c> high bandwidth comments: use makefile
> from codespell to install, changing prefix a la
> https://github.com/mr-c/misc/wiki/Linux-Home-Directory-Program-Installation
>
> e.g. make --prefix=$HOME install
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1203#issuecomment-125753394>.
>
-- 
Michael R. Crusoe: Programmer & Bioinformatician crusoe@ucdavis.edu
The lab for Data Intensive Biology; University of California, Davis
https://impactstory.org/MichaelRCrusoe http://twitter.com/biocrusoe

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
retest this please

File: doc/dev/coding-guidelines-and-review.rst [hunk 1]
```diff
@@ -29,8 +29,36 @@ indentation can be set with::
...
+`export` line to your `.bashrc` or equivalent.
+
+To run codespell over only what has been changed on the branch `my-branch`::
+
+        git diff master..my-branch > diff_file
+        codespell diff_file
+
+To run codespell over a single file::
+
+        codespell path/to/file
```
Comment (mr-c) [inline doc/dev/coding-guidelines-and-review.rst]: Mention `-w` to write the changes in place?

File: doc/dev/coding-guidelines-and-review.rst [hunk 2]
```diff
@@ -29,8 +29,40 @@ indentation can be set with::
 For Python, `PEP 8 <http://www.python.org/dev/peps/pep-0008/>`__ is our
 standard. The ```pep8``` and ```autopep8``` Makefile targets are helpful. 
 
-Code, scripts, and documentation must have its spelling checked. Vim users can
-run::
+Code, scripts, and documentation must have their spelling checked. 
+
+Python-based `codespell` is the preffered tool. `codespell` can be installed
```
Comment (luizirber) [inline doc/dev/coding-guidelines-and-review.rst]: preferred

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
@luizirber Merge, please...?

File: doc/dev/coding-guidelines-and-review.rst [hunk 3]
```diff
@@ -29,8 +29,40 @@ indentation can be set with::
 For Python, `PEP 8 <http://www.python.org/dev/peps/pep-0008/>`__ is our
 standard. The ```pep8``` and ```autopep8``` Makefile targets are helpful. 
 
-Code, scripts, and documentation must have its spelling checked. Vim users can
-run::
+Code, scripts, and documentation must have their spelling checked. 
+
+Python-based `codespell` is the preferred tool. `codespell` can be installed
```
Comment (mr-c) [inline doc/dev/coding-guidelines-and-review.rst]: Now that I understand codespell better, I disagree with it being 'preferred' :-/

Comment (bocajnotnef) [inline doc/dev/coding-guidelines-and-review.rst]: Heh. Fair.

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
@mr-c Ready for merge

File: doc/dev/coding-guidelines-and-review.rst [hunk 4]
```diff
@@ -29,8 +29,40 @@ indentation can be set with::
...
+can be installed via the following::
+
+        mkdir ~/bin
+        git clone git@github.com:lucasdemarchi/codespell.git
+        cd codespell
+        make prefix=${HOME} install
+        export PATH=$PATH:~/bin/
+
+Note, if you want codespell to always be available you will need to add the
+`export` line to your `.bashrc` or equivalent.
```
Comment (mr-c) [inline doc/dev/coding-guidelines-and-review.rst]: `${HOME}\.bashrc`

**Comment by mr-c** [comment]
Otherwise this LGTM; please fix and make mergeable and self-merge

**Pull request merged**